pilot/model/service.go

```diff
@@ -64,6 +64,15 @@ type Service struct {
 	LoadBalancingDisabled bool `json:"-"`
 }
 
+// AuthMigrationPort represents an alias port for a service port that would
+// be used for authentication migration.
+// As of speaking, this information is extracted from service annotations.
+type AuthMigrationPort struct {
+	Port                 int                              `json:"port,omitempty"`
+	AuthenticationPolicy proxyconfig.AuthenticationPolicy `json:"policy,omitempty"`
+	Active               bool                             `json:"active,omitempty"`
+}
```

Comment:
@diemtvu bit confused. if user is explicitly opening up a new port on the pod for auth migration, then isn't it much more simpler to use standard routing rules and point to the new port? Istio routing rules support port based destinations. 

```yaml
destination:
 name: foo
 port : 80
route:
  - destination:
        port: 80
     weight: 90
  - destination:
        port: 9080
     weight: 10
```

This feature can be used for other scenarios as well. This  is currently not implemented but supported in the spec. So, if you implement this, you can orchestrate auth migration easily. 

The solution you have has some issues: you are doing a 100% shift immediately. If something breaks, there is no way to fallback. so its not migration. Its a complete switch with no fallback, unless I change my service port definition.

I am also not seeing in the golden files, how the migration port (Specified in the service declaration with auth policy = true) manifests..